[RN 0.75 new arch] 'utils/RectUtil.h' file not found

### Description

Hi team,

I've been testing upgrading RN 0.74 to 0.75-RC7 with basic stack including `react-navigation` and `react-native-screens`.
The iOS build fails for me with the following error:

```
In file included from /Users/jkosecki/repos/custom/ReactNativeUpgradeTester/node_modules/react-native-screens/ios/RNSScreen.mm:15:
/Users/jkosecki/repos/custom/ReactNativeUpgradeTester/node_modules/react-native-screens/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h:10:10: fatal error: 'utils/RectUtil.h' file not found
#include "utils/RectUtil.h"
```

I have applied the following patch locally (included in the reproduction repo):
```
diff --git a/node_modules/react-native-screens/RNScreens.podspec b/node_modules/react-native-screens/RNScreens.podspec
index be7cc24..14cd2e4 100644
--- a/node_modules/react-native-screens/RNScreens.podspec
+++ b/node_modules/react-native-screens/RNScreens.podspec
@@ -27,7 +27,7 @@ Pod::Spec.new do |s|
     s.subspec "common" do |ss|
       ss.source_files         = ["common/cpp/**/*.{cpp,h}", "cpp/**/*.{cpp,h}"]
       ss.header_dir           = "rnscreens"
-      ss.pod_target_xcconfig  = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/common/cpp\"" }
+      ss.pod_target_xcconfig  = { "HEADER_SEARCH_PATHS" => "\"$(PODS_TARGET_SRCROOT)/common/cpp/react/renderer/components/\"" }
     end
   end
 
```
but I'm not sure if that's correct way to fix this issue.

### Steps to reproduce

1. Remove `patches/react-native-screens+3.34.0.patch`
2. Run `yarn setup && yarn ios`

### Snack or a link to a repository

https://github.com/jankosecki/react-native-upgrade-tester/tree/rn-0.75

### Screens version

3.34.0

### React Native version

0.75-RC7

### Platforms

iOS

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Fabric (New Architecture)

### Build type

Debug mode

### Device

None

### Device model

_No response_

### Acknowledgements

Yes

--- a/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h
+++ b/common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h
@@ -5,9 +5,9 @@
 #endif
 #include <react/debug/react_native_assert.h>
 #include <react/renderer/components/rnscreens/Props.h>
+#include <react/renderer/components/rnscreens/utils/RectUtil.h>
 #include <react/renderer/core/ConcreteComponentDescriptor.h>
 #include "RNSScreenShadowNode.h"
-#include "utils/RectUtil.h"
 
 namespace facebook {
 namespace react {

